content/hardware/08.edu/solution-and-kits/alvik/tutorials/user-manual/user-manual.md

```diff
@@ -112,133 +135,1280 @@ Alvik has five different sensors, all connected to the STM32 and accessible thro
+
+```python
+from arduino_alvik import ArduinoAlvik
+from time import sleep_ms
+import network
+import espnow
+
+# Initialize ESP-NOW
+wlan = network.WLAN(network.STA_IF)
+wlan.active(True)
+esp = espnow.ESPNow()
+esp.active(True)
+
+# Define the peer's MAC address
+peer_mac = b'\x74\x4d\xbd\x00\x00\x00'  # MAC address of the receiver Alvik
+
+# Try to add the peer, and handle the error if the peer already exists
+try:
+    esp.add_peer(peer_mac)
+except OSError as e:
+    if e.args[0] == -12395:  # ESP_ERR_ESPNOW_EXIST
+        print("Peer already exists.")
+    else:
+        raise
+
+# Initialize Alvik
+alvik = ArduinoAlvik()
+alvik.begin()
+
+# Define button to command mapping
+button_commands = {
+    "UP": "MOVE_FORWARD",
+    "DOWN": "MOVE_BACKWARD",
+    "LEFT": "TURN_LEFT",
+    "RIGHT": "TURN_RIGHT",
+    "OK": "STOP",
+    "CANCEL": "STOP",
+    "CENTER": "ACTION"
+}
+
+def send_command(command):
+    esp.send(peer_mac, command.encode())
+    print(f"Sent command: {command}")
+
+print("Controller running. Press buttons to send commands.")
+
+while True:
+    try:
+        if alvik.get_touch_any():
+            if alvik.get_touch_up():
+                send_command(button_commands["UP"])
+            if alvik.get_touch_down():
+                send_command(button_commands["DOWN"])
+            if alvik.get_touch_left():
+                send_command(button_commands["LEFT"])
+            if alvik.get_touch_right():
+                send_command(button_commands["RIGHT"])
+            if alvik.get_touch_ok():
+                send_command(button_commands["OK"])
+            if alvik.get_touch_cancel():
+                send_command(button_commands["CANCEL"])
+            if alvik.get_touch_center():
+                send_command(button_commands["CENTER"])
+
+        sleep_ms(100)
+    except KeyboardInterrupt:
+        alvik.stop()
+        sys.exit()
+```
+
+### Receiver
+
+1. **Initializing ESP-NOW**: The code sets up the ESP32 as a Wi-Fi station and initializes ESP-NOW.
+2. **Callback Function**: A callback function `on_receive_msg` is defined to handle incoming messages. This function decodes the message and moves the robot based on the received command.
+3. **Registering the Callback**: The callback function is registered with ESP-NOW to ensure it gets called when a message is received.
+
+### Sender
+
+1. **Initializing ESP-NOW**: The code sets up the ESP32 as a Wi-Fi station and initializes ESP-NOW.
+2. **Adding a Peer**: The MAC address of the receiver (Alvik robot) is added as a peer.
+3. **Sending Commands**: The function `send_command` sends commands to the receiver. Example commands are provided to move the robot in different directions.
+
+By following these steps, you'll be able to control your Arduino Alvik robot remotely using ESP-NOW, a fast and efficient communication protocol.
+Feel free to tinker with this feature to add new and interesting interactions.
 
 ## Maintenance

```

Comment:
Couple of things in this section:
1. Please let's get rid of the "Check wheel alignment" image XD and replace it for another one, render, IDK something better looking 
2. 1st snippet on the "Calibrate Color Sensors" is incomplete or has bugs

Reply:
This is from the original document I will have to request a visual for this and re-test